client/src/store/new-request-module.ts

```diff
@@ -2504,8 +2518,103 @@ export class NewRequestModule extends VuexModule {
     }
     this.mutateDisplayedComponent('ExistingRequestEdit')
   }
+  @Action
+  async parseExactNames (json: { names: [string] }) {
+    let nameObjs = json.names
+    let names = []
+    for (let i = 0; i < nameObjs.length; i++) {
+      names.push({ name: `${nameObjs[i]['name']}`, type: 'exact' })
+    }
+    return names
+  }
+  @Action
+  async parseSynonymNames (json: { names: [string], exactNames: [{ name: string, type: string }] }) {
+    let duplicateNames = []
+    for (let i = 0; i < json.exactNames.length; i++) {
+      duplicateNames.push(json.exactNames[i].name)
+    }
+    let nameObjs = json.names
+    let names = []
+    for (let i = 0; i < nameObjs.length; i++) {
+      if (nameObjs[i]['name_info']['id']) {
+        let name = nameObjs[i]['name_info']['name']
+        if (!duplicateNames.includes(name)) {
+          names.push({ name: name, type: 'synonym' })
+        }
+      }
+    }
+    return names
+  }
+  @Action
+  async getQuickSearch (cleanedName: {exactMatch: string, synonymMatch: string}) {
+    try {
+      this.mutateDisplayedComponent('QuickSearchPending')
+      let encodedAuth = btoa(`${window['quickSearchPublicId']}:${window['quickSearchPublicSecret']}`)
+      const tokenResp = await axios.post(window['authTokenUrl'], 'grant_type=client_credentials', {
+        headers: { Authorization: `Basic ${encodedAuth}`, 'content-type': 'application/x-www-form-urlencoded' }
+      })
+      let token = tokenResp.data.access_token
+      const exactResp = await axios.get('/exact-match?query=' + cleanedName.exactMatch, {
+        headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+      })
+      const synonymResp = await axios.get('/requests/synonymbucket/' + cleanedName.synonymMatch + '/*', {
+        headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+      })
+
+      const exactNames = await this.parseExactNames(exactResp.data)
+      // pass in exactNames so that we can check for duplicates
+      synonymResp.data.exactNames = exactNames
+      const synonymNames = await this.parseSynonymNames(synonymResp.data)
+      this.mutateQuickSearchNames(exactNames.concat(synonymNames))
+      // check if they skipped
+      if (this.quickSearch) {
+        this.mutateDisplayedComponent('QuickSearchResults')
+      }
+    } catch (error) {
+      // handle silently and move on to detailed search
+      console.error('quick search failed: ', error) // eslint-disable-line no-console
```

Comment:
This is not silent: it will get picked up by Sentry. Maybe use console.log or console.warn instead.

Reply:
I want sentry to pick this up, but I don't want the user to see it

Reply:
that's what I meant by silent. I don't think we should ever have a truly silent error

Reply:
In that case, I suggest a more detailed comment, and perhaps add the method name to the error like the other ones in this module do.

Reply:
done

Reply:
It's perfect.